synapse/handlers/room_member.py

```diff
@@ -39,6 +38,11 @@
 logger = logging.getLogger(__name__)
 
 id_server_scheme = "https://"
+ROOM_COMPLEXITY_TOO_GREAT = (
+    "Your homeserver is unable to join rooms this large or complex. "
+    "Please speak to your server administrator, or upgrade your instance "
+    "to join this room."
+)
```

Comment:
I don't know if we want to make this configurable? The wording might want to change depending on why admins have added the limit?

Reply:
sure, done